jdaviz/configs/specviz/plugins/line_analysis/line_analysis.py

```diff
@@ -285,9 +323,49 @@ def _calculate_statistics(self, *args, **kwargs):
                 # TODO: update specutils to be consistent with region vs regions and default to
                 # regions=None so this elif can be removed
                 temp_result = FUNCTIONS[function](spec_subtracted, region=None)
+                self.results_centroid = temp_result
             else:
                 temp_result = FUNCTIONS[function](spec_subtracted)
 
             temp_results.append({'function': function, 'result': str(temp_result)})
 
         self.update_results(temp_results, mark_x, mark_y)
+
+        if not self.selected_line and self.identified_line:
+            # default to the identified line
+            self.selected_line = self.identified_line
+
```

Comment:
I think you need one more check here to run `_compute_redshift_for_selected_line()` if a line has been selected. Currently if you have a line selected and then go run the line analysis function for the first time, the calculated redshift remains 0 until you select a different line.